tests/client/test_search_and_parse_results.py

```diff
@@ -0,0 +1,104 @@
+from unittest.mock import MagicMock, patch
+
+from lxml import etree
+
+from caselawclient.Client import SearchParameters, api_client
+from caselawclient.models.search_results import SearchResults
+
+
+@patch("caselawclient.Client.api_client.advanced_search")
+def test_perform_advanced_search(mock_advanced_search):
```

Comment:
Taken this from the current open PR in public ui which I will be updating once this PR is merged in and new api client is released:

https://github.com/nationalarchives/ds-caselaw-public-ui/pull/701/files#diff-95daf164b3b15f1fb2437325b5b44436775eaa1a6c976142426f1247455d3080